contracts/BridgeCommittee.sol

```diff
@@ -61,37 +64,32 @@ contract BridgeCommittee is IBridgeCommittee, CommitteeUpgradeable {
         // Loop over the signatures and check if they are valid
         uint16 approvalStake;
         address signer;
-        // Declare an array to store the recovered addresses
-        address[] memory seen = new address[](signatures.length);
-        uint256 seenIndex = 0;
+        uint256 bitmap;
+
         for (uint16 i = 0; i < signatures.length; i++) {
             bytes memory signature = signatures[i];
             // recover the signer from the signature
             (bytes32 r, bytes32 s, uint8 v) = splitSignature(signature);
 
             (signer,,) = ECDSA.tryRecover(BridgeMessage.computeHash(message), v, r, s);
 
-            // TODO: explore alternative dup checking methods (this is O(n^2)
-            // Check if the address has already been seen
-            bool found = false;
-            for (uint256 j = 0; j < seen.length; j++) {
-                if (seen[j] == signer) {
-                    found = true;
-                    break;
-                }
-            }
-            require(!found, "BridgeCommittee: Duplicate signature, address already seen");
-
-            // Add the address to the array
-            seen[seenIndex++] = signer;
+            require(committeeStake[signer] > 0, "BridgeCommittee: Signer not a committee member");
 
-            // Check if the signer is a committee member and not already approved
-            require(committeeMembers[signer] > 0, "BridgeCommittee: Not a committee member");
+            uint8 index = addressIndex[signer];
+            uint256 mask = 1 << index;
+            if (bitmap & mask == 0) {
+                bitmap |= mask;
+            } else {
+                // Duplicate signature
+                revert("BridgeCommittee: Duplicate signature");
+            }
```

Comment:
do you think it's better to revert than ignore the sig?

Reply:
good point. I reverted to abide by the old test we have, but I think its better to skip like we do with blocklisted sigs.